src/Aspire.Hosting/HttpService/HttpServiceBuilderExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Sockets;
+using System.Text.Json;
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Http Service resources to the application model.
+/// </summary>
+public static class HttpServiceBuilderExtensions
+{
+    /// <summary>
+    /// Adds a Http Service to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="uri"></param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{HttpServiceResource}"/>.</returns>
+    public static IResourceBuilder<HttpServiceResource> AddHttpService(this IDistributedApplicationBuilder builder, string name, string uri)
+    {
+        return builder.AddHttpService(name, new Uri(uri, UriKind.Absolute));
+    }
+
+    /// <summary>
+    /// Adds a Http Service to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="uri"></param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{HttpServiceResource}"/>.</returns>
+    public static IResourceBuilder<HttpServiceResource> AddHttpService(this IDistributedApplicationBuilder builder, string name, Uri uri)
+    {
+        var address = uri.GetLeftPart(UriPartial.Authority).Remove(0, uri.GetLeftPart(UriPartial.Scheme).Length);
```

Comment:
Add a comment why this exists?

Reply:
I wanted to get the URL without its scheme, I'll add a comment and try to add a better implementation for it.